crates/sparrow-physical/src/expr.rs

```diff
@@ -0,0 +1,79 @@
+use std::borrow::Cow;
+
+use arrow_schema::DataType;
+
+/// A physical expression which describes how a value should be computed.
+///
+/// Generally, each expression computes a column of values from zero or more
+/// input columns. Expressions appear in a variety of places within the steps
+/// that make up a physical plan.
+#[derive(Debug, serde::Serialize, serde::Deserialize)]
+
+pub struct Expr {
+    pub kind: ExprKind,
+    /// Arguments to the expression.
+    ///
+    /// These are indices referencing earlier expressions.
+    pub children: Vec<usize>,
+    /// The type produced by the expression.
+    pub result_type: DataType,
+}
+
+#[derive(
+    Clone,
+    Debug,
+    Eq,
+    Hash,
+    Ord,
+    PartialEq,
+    PartialOrd,
+    serde::Serialize,
+    serde::Deserialize,
+    enum_as_inner::EnumAsInner,
+)]
+#[serde(rename_all = "snake_case")]
+pub enum ExprKind {
+    /// Apply the named instruction to the given children.
+    Call(Cow<'static, str>),
+    /// Reference an input column by name.
+    Column(String),
+    /// A boolean literal.
+    BooleanLiteral(bool),
```

Comment:
I was originally going to have a `Literal(ScalarValue)` but that is a very implementation specific option. My current thinking is that it may be better to use a combination of these 3 simple literals (booleans, strings, numerics) and maybe `null` literals, in combination with the `DataType` of the expression to create the scalar value if we need it.

I'd propose we see how it plays out when we go to generate and/or execute these and see what we think. Thoughts?

Reply:
Most types I believe would use some of these -- for instance, `type=TimestampMillis` with a `NumberLiteral` would interpret the number as the number of millis.